### PR TITLE
Return last recorded frame for snapshot endpoint

### PR DESCRIPTION
**Pull request by scraton**
As mentioned in #61, this change returns the last frame for the snapshot endpoint, allowing the program to return the last frame even if the camera is disabled.

It seems this data was already being stored, so just returning a copy of that was all that was necessary. The extra code removals here were done to clean up cruft that was no longer necessary.

### CODE REVIEW

File: mtp/server.go [hunk 1]
```diff
@@ -261,16 +258,11 @@ func (s *LVServer) unregisterMotionClient(w *MJPEGResponseWriter) {
...
-
-	<-writer.Context.Done()
+	var jpeg []byte
+	jpeg = s.copyFrame()
 
-	s.controlLock.Lock()
-	delete(s.snapshotClients, writer)
-	s.controlLock.Unlock()
+	writer := NewSnapshotResponseWriter(w)
+	writer.Write(jpeg)
```
Comment (puhitaku) [inline mtp/server.go:265]: We no longer need a struct `SnapshotResponseWriter`. Let's have all handling done here.

```suggestion
	w.Header().Set("Content-Type", "image/jpeg")
	w.Write(jpeg)
```

**Review by puhitaku — CHANGES_REQUESTED** [review]
I honestly forgot some part of the detail under the hood, but anyway, I've tested the updated one, and it serves the last frame after disconnecting my D5300 🎉 

Two things are left before merging:
1. Please merge the SnapshotResponseWriter into the handler and remove `snapshot.go`.
2. There is an unwanted master-merge commit 3698d01. Please rebase your master branch to the upstream (= my) master branch. Generally speaking, committing your change into a non-master branch is better not to encounter this kind of ambiguation.

**Comment by scraton** [comment]
Cool :+1: Changes applied, let me know if it looks up to par.

**Review by puhitaku — APPROVED** [review]
LGTM

**Comment by puhitaku** [comment]
I've bumped and released [version 1.6.0](https://github.com/puhitaku/mtplvcap/releases/tag/v1.6.0)! Thank you for your contribution!

**Pull request merged**